src/engraving/rendering/dev/tlayout.cpp

```diff
@@ -2312,33 +2312,24 @@ void TLayout::layout(FretCircle* item, LayoutContext& ctx)
     layoutFretCircle(item, ctx, item->mutLayoutData());
 }
 
-void TLayout::layout(Glissando* item, LayoutContext& ctx)
+static void layoutGlissando(const Glissando* item, LayoutContext& ctx, Glissando::LayoutData* ldata)
 {
     double _spatium = item->spatium();
 
-    if (ctx.conf().isPaletteMode() || !item->startElement() || !item->endElement()) {    // for use in palettes or while dragging
-        if (item->spannerSegments().empty()) {
-            item->add(item->createLineSegment(ctx.mutDom().dummyParent()->system()));
-        }
-        LineSegment* s = item->frontSegment();
-        s->setPos(PointF(-_spatium * Glissando::GLISS_PALETTE_WIDTH / 2, _spatium * Glissando::GLISS_PALETTE_HEIGHT / 2));
-        s->setPos2(PointF(_spatium * Glissando::GLISS_PALETTE_WIDTH, -_spatium * Glissando::GLISS_PALETTE_HEIGHT));
-        layout(s, ctx);
-        return;
-    }
-    layoutLine(item, ctx);
+    TLayout::layoutLine(const_cast<Glissando*>(item), ctx);
+
     if (item->spannerSegments().empty()) {
         LOGD("no segments");
         return;
     }
-    item->setPos(0.0, 0.0);
+    ldata->setPos(0.0, 0.0);
 
-    Note* anchor1     = toNote(item->startElement());
-    Note* anchor2     = toNote(item->endElement());
-    Chord* cr1         = anchor1->chord();
-    Chord* cr2         = anchor2->chord();
-    GlissandoSegment* segm1 = toGlissandoSegment(item->frontSegment());
-    GlissandoSegment* segm2 = toGlissandoSegment(item->backSegment());
+    Note* anchor1 = toNote(item->startElement());
+    Note* anchor2 = toNote(item->endElement());
+    Chord* cr1 = anchor1->chord();
+    Chord* cr2 = anchor2->chord();
+    GlissandoSegment* segm1 = toGlissandoSegment(const_cast<Glissando*>(item)->frontSegment());
```

Comment:
This is getting a little bit awful with all those const_cast... and it brings us to the difficult question of what to do with spanner segments. We can leave it like this for now, or we can make the `Glissando*` not `const`; both are not ideal of course, but it's temporary anyway.

Reply:
At the moment `const_cast` indicate problems places, this is hack. We should think about each `const_cast`, but I think will be next iteration of work, because now it just technikaly changes, without changes algoritm. I would not like mix tech and algo changes.